### PR TITLE
fixes #15197 - find hypervisors by uuid before name

### PR DESCRIPTION
**Pull request by thomasmckay**

### CODE REVIEW

File: app/lib/actions/katello/host/hypervisors_update.rb [hunk 1]
```diff
@@ -29,18 +29,19 @@ def update_or_create_hypervisor(environment, content_view, hypervisor_json)
...
               name = duplicate_name
+              host = nil
             end
           elsif ::Host.find_by(:name => name)
             name = duplicate_name
           end
 
           host = ::Katello::Host::SubscriptionFacet.find_or_create_host_for_hypervisor(name,
-                                    content_view.organization)
+                                    content_view.organization) unless host
```
Comment (ehelms) [inline app/lib/actions/katello/host/hypervisors_update.rb]: Please no multi-line trailing conditionals

File: app/lib/actions/katello/host/hypervisors_update.rb [hunk 2]
```diff
@@ -61,6 +62,11 @@ def update_or_create_hypervisor(environment, content_view, hypervisor_json)
 
           host.save!
         end
+
+        def find_host_by_uuid_or_name(hypervisor_json)
+          facet = ::Katello::Host::SubscriptionFacet.find_by(:uuid => hypervisor_json[:uuid])
+          facet.nil? ? ::Host.find_by(:name => hypervisor_json[:name]) : facet.host
```
Comment (ehelms) [inline app/lib/actions/katello/host/hypervisors_update.rb]: This logic seems to make https://github.com/Katello/katello/pull/6076/files#diff-bf897becee6d218f2e9b589c5f66dcfdR39 redundant in terms of database queries

**Comment by thomasmckay** [comment]
marked as WIP as I need to fix/add tests

**Comment by thomasmckay** [comment]
@ehelms @jlsherrill - Please review when you are able.

**Comment by jlsherrill** [comment]
ACK from me

**Comment by ehelms** [comment]
:+1:

**Pull request merged**